stdlib/array.ml

```diff
@@ -30,6 +30,12 @@ external unsafe_blit :
 external create_float: int -> float array = "caml_make_float_vect"
 let make_float = create_float
 
+module Floatarray = struct
```

Comment:
In the naming side, I don't like `Array.Floatarray`.  I'd prefer any of `Array.FloatArray`, `Array.Float, `Float.Array`, or a toplevel `FloatArray`.

Reply:
@yallop `Array.Float` has my preference, and I'd say it's just as easy to extend.

Reply:
@damiendoligez: I think type-specialized array representation is a bit like type-specialized `compare` in this respect.  Rather than enumerating all the type-specialized `compare` functions in a single module, we've put each instance alongside its type: `Int64.compare`, `Char.compare`, etc.

Doing things that way has the advantage that it extends well to types outside the standard library, too.

Reply:
Additionally, it would avoid awkward dependency cycles. Array is towards the start of the dependency chain right now. Adding `Array.*` for each base type of the stdlib would need to move it towards the end.

Reply:
I think `Float.Array` is most satisfactory.  It matches up with existing usage in the compiler of `Foo.Map`, `Foo.Set`, etc.